packages/react-core/src/components/ExpandableSection/ExpandableSection.tsx

```diff
@@ -80,9 +81,8 @@ export class ExpandableSection extends React.Component<ExpandableSectionProps, E
     // uncontrolled
     if (isExpanded === undefined) {
       propOrStateIsExpanded = this.state.isExpanded;
-      onToggle = () => {
-        onToggleProp();
-        this.setState(prevState => ({ isExpanded: !prevState.isExpanded }));
+      onToggle = isOpen => {
+        this.setState({ isExpanded: isOpen }, () => onToggleProp(this.state.isExpanded));
```

Comment:
We should only do this is `isOpen/isExpanded` is provided, otherwise we should what we were doing before.